nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java

```diff
@@ -290,32 +285,34 @@ public void renameTable(TableIdentifier from, TableIdentifier toOriginal) {
   @Override
   public void createNamespace(Namespace namespace, Map<String, String> metadata) {
     try {
-      reference.checkMutable();
-      api.createNamespace()
-          .reference(reference.getReference())
+      client.getRef().checkMutable();
+      client.getApi().createNamespace()
+          .reference(client.getRef().getReference())
           .namespace(org.projectnessie.model.Namespace.of(namespace.levels()))
           .create();
-      refresh();
+      client.refresh();
     } catch (NessieNamespaceAlreadyExistsException e) {
       throw new AlreadyExistsException(e, "Namespace '%s' already exists.", namespace);
     } catch (NessieNotFoundException e) {
-      throw new RuntimeException(String.format("Cannot create Namespace '%s': ref is no longer valid.", namespace), e);
+      throw new RuntimeException(String.format("Cannot create Namespace '%s': " +
+          "ref '%s' is no longer valid.", namespace, client.getRef().getName()), e);
```

Comment:
Can this be moved into the Iceberg client?

Reply:
yep, done